File: widgets/label.lua
local _context = getContext()
local _painter = _context:getPainter()
local _style = _painter:getStyle()

exports('label', function(text)
	_painter:setText(text)
	_painter:setTextOpts()

	local w = _context:getWidgetWidth() or _painter:calculateTextWidth()
	local h = _style.widget.height

	_context:beginDraw(w, h)

	_painter:setColor(_style.color.secondary)
	_painter:drawText(_style.label.text.offset)

	_context:endDraw()
end)

File: widgets/label.js
exports('label', function (text) {
	painter.setText(text);
	painter.setTextOpts();

	const w = context.getWidgetWidth() || painter.calculateTextWidth();
	const h = style.widget.height;

	context.beginDraw(w, h);

	painter.setColor(style.color.secondary);
	painter.drawText(style.label.text.offset);

	context.endDraw();
});
